### PR TITLE
Fix #568, avoid redirect when joining member

### LINKED ISSUE #568: Multi master server setup failure
Hi, Chirslusf. I have encountered a problem when setting up a multi master server cluster according the wiki. I have 3 vm:

```
vm1: 10.100.35.21
vm2: 10.100.35.20 
vm3: 10.100.35.22
```

I start weed master server one by one in below order:

```
vm1: ./weed master  -ip 10.100.35.21 -port=9333 -mdir=./master
vm2: ./weed master -ip 10.100.35.20 -port=9333 -mdir=./master  -peers=10.100.35.21:9333
vm3: ./weed master -ip 10.100.35.22 -port=9333 -mdir=./master  -peers=10.100.35.20:9333
```

And then we can see  the log output of each start:

```
vm1:
# ./weed master  -ip 10.100.35.21 -port=9333 -mdir=./master
I0927 15:13:02 14610 file_util.go:20] Folder ./master Permission: -rwxr-xr-x
I0927 15:13:02 14610 master_server.go:62] Volume Size Limit is 30000 MB
I0927 15:13:02 14610 master.go:87] Start Seaweed Master 0.76 at 0.0.0.0:9333
I0927 15:13:02 14610 raft_server.go:56] Peers Change: [10.100.35.21:9333] => []
I0927 15:13:02 14610 raft_server.go:98] Initializing new cluster
I0927 15:13:02 14610 master_server.go:95] [ 10.100.35.21:9333 ] I am the leader!


I0927 15:13:12 14610 raft_server_handlers.go:16] Processing incoming join. Current Leader 10.100.35.21:9333 Self 10.100.35.21:9333 Peers map[]
I0927 15:13:12 14610 raft_server_handlers.go:20] Command:{"name":"10.100.35.20:9333","connectionString":"http://10.100.35.20:9333"}
I0927 15:13:12 14610 raft_server_handlers.go:27] join command from Name 10.100.35.20:9333 Connection http://10.100.35.20:9333


vm2:

# ./weed master -ip 10.100.35.20 -port=9333 -mdir=./master  -peers=10.100.35.21:9333
I0927 15:13:11  3117 file_util.go:20] Folder ./master Permission: -rwxr-xr-x
I0927 15:13:11  3117 master_server.go:62] Volume Size Limit is 30000 MB
I0927 15:13:11  3117 master.go:87] Start Seaweed Master 0.76 at 0.0.0.0:9333
I0927 15:13:11  3117 raft_server.go:56] Peers Change: [10.100.35.20:9333] => [10.100.35.21:9333]
I0927 15:13:11  3117 raft_server.go:78] Joining cluster: 10.100.35.21:9333
I0927 15:13:12  3117 raft_server.go:166] Attempting to connect to: http://10.100.35.21:9333/cluster/join
I0927 15:13:12  3117 raft_server.go:211] Post returned status:  200


I0927 15:13:52  3117 raft_server_handlers.go:16] Processing incoming join. Current Leader 10.100.35.21:9333 Self 10.100.35.20:9333 Peers map[10.100.35.21:9333:0xc42021fa40]
I0927 15:13:52  3117 raft_server_handlers.go:20] Command:{"name":"10.100.35.22:9333","connectionString":"http://10.100.35.22:9333"}
I0927 15:13:52  3117 raft_server_handlers.go:27] join command from Name 10.100.35.22:9333 Connection http://10.100.35.22:9333
I0927 15:13:52  3117 raft_server_handlers.go:47] Redirecting to 301 http://10.100.35.21:9333/cluster/join

vm3:
# ./weed master -ip 10.100.35.22 -port=9333 -mdir=./master  -peers=10.100.35.20:9333
I0927 15:13:51  2964 file_util.go:20] Folder ./master Permission: -rwxr-xr-x
I0927 15:13:51  2964 master_server.go:62] Volume Size Limit is 30000 MB
I0927 15:13:51  2964 master.go:87] Start Seaweed Master 0.76 at 0.0.0.0:9333
I0927 15:13:51  2964 raft_server.go:56] Peers Change: [] => [10.100.35.20:9333]
I0927 15:13:51  2964 raft_server.go:78] Joining cluster: 10.100.35.20:9333
I0927 15:13:52  2964 raft_server.go:166] Attempting to connect to: http://10.100.35.20:9333/cluster/join
I0927 15:13:52  2964 raft_server.go:211] Post returned status:  404 404 page not found

I0927 15:13:52  2964 raft_server.go:171] Post returned error:  404 page not found

I0927 15:13:52  2964 raft_server.go:82] No existing server found. Starting as leader in the new cluster.
I0927 15:13:52  2964 master_server.go:95] [ 10.100.35.22:9333 ] I am the leader!

```

It seems that the weed master on vm3 did not handle the 301 redirect to connect to the master(current leader) on vm1. 

Is this a bug? looking forward to your reply. 3ks.

### PR DESCRIPTION
**Pull request by xwjdsh**
When adding a new member, query it's leader firstly, then send join request to the leader server directly, this could avoid occurring redirect in some cases.

### CODE REVIEW

**Comment by chrislusf** [comment]
I wonder why this is needed. In what case this would be helpful?

File: weed/compress/delta_binary_pack32.go [hunk 1]
```diff
@@ -1,8 +1,8 @@
 package compress
 
 import (
-	"github.com/reducedb/encoding/cursor"
-	"github.com/reducedb/encoding/delta/bp32"
+	"github.com/dataence/encoding/cursor"
```
Comment (chrislusf) [inline weed/compress/delta_binary_pack32.go:4]: is this related?

**Comment by xwjdsh** [comment]
@chrislusf Sorry, this is my fault.

I probably used the latest released version for debugging #568 , so the error always exists, then I have an idea that avoids redirecting. I tested master branch just now, the redirect work fine.

Please feel free to close this PR and forgive me waste your time.

File: weed/compress/delta_binary_pack32.go (hunk 1 above)
Comment (xwjdsh) [reply weed/compress/delta_binary_pack32.go:4]: The diff for fixing CI failed, the package changed its name in upstream, so `go get ./weed/...` command gets an error in CI. Maybe we should use glide to fetch dependence in CI.
